### PR TITLE
Nextflow scripts for export

### PR DESCRIPTION
**Pull request by kbab**
* Fix minor bug due to missing page_size parameter in using api_client
* Create a nextflow script to run export of study, dataset and image metadata one accession ID at a time and an associated python script to merge the resulting jsons into one. See [https://app.clickup.com/t/86975pvwc](https://app.clickup.com/t/86975pvwc).

### CODE REVIEW

File: bia-export/bia_export/website_export/images/retrieve.py [hunk 1]
```diff
@@ -25,12 +26,14 @@ def retrieve_images(
 
     else:
         dataset_list: List[api_models.Dataset] = api_client.get_dataset_linking_study(
-            str(context.study_uuid)
+            str(context.study_uuid),
+            page_size=DEFAULT_PAGE_SIZE,
```
Comment (AybukeKY) [inline bia-export/bia_export/website_export/images/retrieve.py:30]: This is assuming the number of datasets in a study won't be larger than default_page_size parameter. Works for 500 at the moment, but it'd be good to make sure all datasets are retrieved.

File: bia-export/bia_export/website_export/images/retrieve.py [hunk 2]
```diff
@@ -25,12 +26,14 @@ def retrieve_images(
...
         )
         api_images = []
         for dataset in dataset_list:
             api_images += get_all_api_results(
-                dataset.uuid, api_client.get_image_linking_dataset
+                dataset.uuid,
+                api_client.get_image_linking_dataset,
```
Comment (AybukeKY) [inline bia-export/bia_export/website_export/images/retrieve.py:36]: I think this also needs to have a page size parameter? If so it'd need to loop or set the page size such that all images of the dataset are retrieved.

Comment (kbab) [reply bia-export/bia_export/website_export/images/retrieve.py:36]: In this case get_all_api_results is a wrapper function that calls the specified api client function. It has a default page size setting (of 20) and polls until all data received

File: bia-export/bia_export/website_export/images/retrieve.py (hunk 1 above)
Comment (kbab) [reply bia-export/bia_export/website_export/images/retrieve.py:30]: Yes - the above is a quick fix. Default can be increased if necessary

File: bia-export/nextflow/set_local_env.sh_template [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Configuration for Nextflow running bia-export
+
+# For bia-integrator-api
+export BIA_API_BASEPATH="https://wwwdev.ebi.ac.uk/bioimage-archive/api"
+export BIA_API_USERNAME=''
+export BIA_API_PASSWORD=''
+export BIA_USERNAME=''
```
Comment (AybukeKY) [inline bia-export/nextflow/set_local_env.sh_template:7]: What is BIA_USERNAME and password used for as there's already a BIA_API_USERNAME?

Comment (kbab) [reply bia-export/nextflow/set_local_env.sh_template:7]: This is a carry over of change in naming convention. Only one version is needed - will resolve which one and deleted unneeded.

Comment (kbab) [reply bia-export/nextflow/set_local_env.sh_template:7]: Looked at the code and actually they are not required as we are only reading using the public api. Will delete lines 4-7!

File: bia-export/nextflow/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+1. Navigate to your working directory (where `nextflow.config` was copied to)
+1. Create a manifest of the form study_uuid:accession_id and save as `./tmp/manifest.txt`. E.g.
+```
+00357b15-5a36-48e3-ab9d-9af5b697afef:S-BSST445
+0077e2c7-567a-46fa-8848-f39a86894755:S-BIAD851
+0086e8f0-8c99-4fd2-8971-833c147443c8:S-BIAD821
+00da4466-84cd-484e-adf5-2520c9c8ff0c:S-BIAD465
+```
+2. Set up the local environment: `source ./set_local_env.sh`
+3. Run export: `nextflow run $BIA_EXPORT_DIR/nextflow/run_bia_export.nf`
```
Comment (AybukeKY) [inline bia-export/nextflow/README.md:22]: I got "WARN: Access to undefined parameter `manifest_path` -- Initialise it to a default value eg. `params.manifest_path = some_value`
Missing `fromPath` parameter"

Comment (kbab) [reply bia-export/nextflow/README.md:22]: I am guessing  nextflow cannot find the configuration file. In your case is nextflow.config in the same directory as you are running the command from? I will amend readme to include `-c path_to_nextflow_config` to override this assumption.

**Review by AybukeKY — APPROVED** [review]
LGTM. Would be good to mention that one needs to install the bia-export package via "poetry install" first and then run the workflow.

**Pull request merged**